actool/manifest.go

```diff
@@ -0,0 +1,364 @@
+}
+
+func getIsolatorStr(name, value string) string {
+	return fmt.Sprintf(`
+                {
+                    "name": "%s",
+                    "value": { %s }
+                }`, name, value)
+}
+
+func patchManifest(im *schema.ImageManifest) error {
+
+	if patchName != "" {
+		name, err := types.NewACName(patchName)
+		if err != nil {
+			return err
+		}
+		im.Name = *name
+	}
+
+	if patchCaps != "" {
+		app := im.App
+		if app == nil {
+			return fmt.Errorf("no app in the manifest")
+		}
+		isolator := app.Isolators.GetByName(types.LinuxCapabilitiesRetainSetName)
+		if isolator != nil {
+			return fmt.Errorf("isolator already exists")
+		}
+
+		capsList := strings.Split(patchCaps, ",")
+		caps := fmt.Sprintf(`"set": ["%s"]`,
+			strings.Join(capsList, `", "`))
+		isolatorStr := getIsolatorStr(types.LinuxCapabilitiesRetainSetName,
+			caps)
+		isolator = &types.Isolator{}
+		err := isolator.UnmarshalJSON([]byte(isolatorStr))
+		if err != nil {
+			return err
+		}
+		app.Isolators = append(app.Isolators, *isolator)
+	}
+
+	if patchMounts != "" {
+		app := im.App
+		if app == nil {
+			return fmt.Errorf("no app in the manifest")
+		}
+		mounts := strings.Split(patchMounts, ":")
+		for _, m := range mounts {
+			mountPoint, err := types.MountPointFromString(m)
+			if err != nil {
+				return fmt.Errorf("cannot parse mount point %q", m)
+			}
+			app.MountPoints = append(app.MountPoints, *mountPoint)
+		}
+	}
+	return nil
+}
+
+// iterate iterates over the tar reader and optionally write in the tar writer if not nil
+func iterate(tr *tar.Reader, tw *tar.Writer, printManifest bool, newManifest []byte) error {
+Tar:
+	for {
+		hdr, err := tr.Next()
+		switch err {
+		case io.EOF:
+			break Tar
+		case nil:
+			if filepath.Clean(hdr.Name) == aci.ManifestFile {
+				var new_bytes []byte
+
+				bytes, err := ioutil.ReadAll(tr)
+				if err != nil {
+					return err
+				}
+
+				if printManifest && !catPrettyPrint {
+					fmt.Println(string(bytes))
+				}
+
+				im := &schema.ImageManifest{}
+				err = im.UnmarshalJSON(bytes)
+				if err != nil {
+					return err
+				}
+
+				if printManifest && catPrettyPrint {
+					output, err := json.MarshalIndent(im, "", "    ")
+					if err != nil {
+						return err
+					}
+					fmt.Println(string(output))
+				}
+
+				if tw == nil {
+					return nil
+				}
+
+				if len(newManifest) == 0 {
+					err = patchManifest(im)
+					if err != nil {
+						return err
+					}
+
+					new_bytes, err = im.MarshalJSON()
+					if err != nil {
+						return err
+					}
+				} else {
+					new_bytes = newManifest
+				}
+
+				hdr.Size = int64(len(new_bytes))
+				err = tw.WriteHeader(hdr)
+				if err != nil {
+					return err
+				}
+
+				_, err = tw.Write(new_bytes)
+				if err != nil {
+					return err
+				}
+			} else if tw != nil {
+				err := tw.WriteHeader(hdr)
+				if err != nil {
+					return err
+				}
+				_, err = io.Copy(tw, tr)
+			}
+			if err != nil {
```

Comment:
inconsistency nit, weird to handle error here for the second branch but on line 188 for the first branch